atmo/jobs/models.py

```diff
@@ -226,3 +237,32 @@ def delete(self, *args, **kwargs):
 
     def get_results(self):
         return self.provisioner.results(self.identifier, self.is_public)
+
+
+@python_2_unicode_compatible
+class SparkJobHistory(EditedAtModel):
+
+    spark_job = models.ForeignKey(
+        SparkJob,
+        on_delete=models.CASCADE,
+        related_name='history',
+        related_query_name='history',
+    )
+    jobflow_id = models.CharField(
+        max_length=50,
+        blank=True,
+        null=True,
+    )
+    run_date = models.DateTimeField(
+        blank=True,
+        null=True,
+        help_text="Date/time that the job was started.",
+    )
+    most_recent_status = models.CharField(
```

Comment:
I started with that too. But since this gets updated whenever the rq job runs and updates all the statuses of the jobs, it will change as the lifecycle of the cluster changes. So it really is the most recent status of this particular job run.